src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -239,6 +240,20 @@ module.exports = function<T, P, I, TI, C, CX>(
             currentHostContext,
             workInProgress
           );
+
+          // A component cannot be focused until it's mounted.
+          // Flag components with autoFocus for later work.
+          switch (type) {
+            case 'button':
+            case 'input':
+            case 'select':
+            case 'textarea':
+              if (newProps.autoFocus) {
```

Comment:
Good call! I've updated the description to be more clear. 😁

I thought everyone else was on holiday today!